src/reader.cpp

```diff
@@ -21,9 +21,10 @@ Reader::~Reader()
 
 // ----------------------------------------------------------------------------------------------------
 
-bool Reader::read(const std::string& name)
+bool Reader::read(const std::string& name, const NodeType type, const RequiredOrOptional opt)
```

Comment:
Please add a doc string describing the intended semantics of the function and its parameters.

With pure guessing it looks like a generic `read` function that can read anything, however there do not seem to be changes that read anything else than a map or a dict, so maybe it's less generic?
If so, there are also interface functions for reading a group and map, which would suggest this function is internal (ie non-public)?